### PR TITLE
Upgrade Sentry, Requests, Boto3 to latest version

### PR DESCRIPTION
**Pull request by AmitPhulera**
## Product Description

When Dependabot raised a PR to update Sentry tests failed because a new deprecation warning was introduced in the code. This new deprecation was added by `urllib3` in their version `1.26.12`.

Our version of Boto3 was using this deprecated module and hence we were seeing the warning.

In the process of figuring out the cause, I came across https://github.com/urllib3/urllib3/issues/2680 which suggested us to get on a newer version of requests as well.

I was pairing with @gherceg yesterday and investigated this issue. On finding the root cause of deprecation we tried upgrading `urllib` to the latest version we came across a new deprecation warning which was coming from the version of elasticsearch that we are using.

So based on our findings I created this PR that
- Introduces a new command in make file to upgrade a specific dependency.
- Updates Sentry, boto3 and requests
- Adds the warning thrown by ES module to ignore list.

Review by commit 🐟 

<hr/>

## Information regarding boto3 update

We tried updating boto3 to the latest version which was 1.26.108 but we realised that tests started failing with 403. A stacktrace from one of the tests -

```

  ERROR: corehq.blobs.tests.test_migrate_compress:TestCompressMigration.test_compress_all
  ----------------------------------------------------------------------
  Traceback (most recent call last):
    File "/mnt/commcare-hq-ro/corehq/blobs/tests/test_migrate_compress.py", line 29, in setUp
      self.db.put(BytesIO(data), meta=meta)
    File "/mnt/commcare-hq-ro/corehq/blobs/s3db.py", line 70, in put
      s3_bucket = self._s3_bucket(create=True)
    File "/mnt/commcare-hq-ro/corehq/blobs/s3db.py", line 156, in _s3_bucket
      self.db.meta.client.head_bucket(Bucket=self.s3_bucket_name)
    File "/usr/local/lib/python3.9/site-packages/botocore/client.py", line 530, in _api_call
      return self._make_api_call(operation_name, kwargs)
    File "/usr/local/lib/python3.9/site-packages/botocore/client.py", line 960, in _make_api_call
      raise error_class(parsed_response, operation_name)
  botocore.exceptions.ClientError: An error occurred (403) when calling the HeadBucket operation: Forbidden

```

But this was only observed in the tests that were running on minio container and not on staging environment that was actually dealing with S3. 

So for the time being we are setting boto3 to a version that is making our tests happy. 

More info on the slack thread https://dimagi.slack.com/archives/C01G688ECF9/p1680878372067559

### Safety story

Would be putting it out on staging for testing.
### QA Plan

Not sure if QA should be requested on this one as the upgrades were compatible with all the dependencies and have gone through changelog where nothing looks suspicious.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by millerdev — APPROVED** [review]

File: requirements/Makefile [hunk 1]
```diff
@@ -17,3 +17,14 @@ upgrade-requirements:
...
+# To upgrade a specific package to latest version use this command
+# make upgrade-package package={package_name}
+upgrade-package: export CUSTOM_COMPILE_COMMAND=`make requirements` or `make upgrade-requirements`
+upgrade-package:
+	pip-compile --upgrade-package $(package) -o requirements.txt requirements.in --allow-unsafe
+	pip-compile --upgrade-package $(package) -o prod-requirements.txt prod-requirements.in --allow-unsafe
+	pip-compile --upgrade-package $(package) -o test-requirements.txt test-requirements.in --allow-unsafe
+	pip-compile --upgrade-package $(package) -o dev-requirements.txt dev-requirements.in --allow-unsafe
+	pip-compile --upgrade-package $(package) -o docs-requirements.txt docs-requirements.in --allow-unsafe
+	python ../scripts/purge-platform-pkgs.py ./*requirements.txt
```
Comment (gherceg) [inline requirements/Makefile:30]: add newline

**Review by gherceg — APPROVED** [review]

File: requirements/Makefile (hunk 1 above)
Comment (AmitPhulera) [reply requirements/Makefile]: [09891ca](https://github.com/dimagi/commcare-hq/pull/32820/commits/09891ca716a628a59c7d2e6e742b1b6b7e6d89d1)

File: requirements/base-requirements.in [hunk 2]
```diff
@@ -1,7 +1,9 @@
 alembic
 architect
 attrs>=21.4
-boto3
+# Updating to a higher version breaks tests running in minio more on https://github.com/dimagi/commcare-hq/pull/32820
+# Can be updated if the tests are resolved
+boto3==1.24.96
```
Comment (millerdev) [inline requirements/base-requirements.in:6]: I'm not sure how aggressively pip-tools or dependabot will attempt to upgrade this library if it is not pinned? If they don't attempt to upgrade it frequently we could drop commit https://github.com/dimagi/commcare-hq/pull/32820/commits/8de1ee44ca1443b76ca520b5d991db8396044f40 and add this comment in the commit message of https://github.com/dimagi/commcare-hq/pull/32820/commits/b91c247a46def6afc7a1f7263290515460e718cb. That way upgrades are not strictly constrained, and this library may naturally upgrade to a newer version where the issue is fixed without us having to explicitly remove the pin.

**Review by millerdev — APPROVED** [review]

File: requirements/base-requirements.in (hunk 2 above)
Comment (AmitPhulera) [reply requirements/base-requirements.in]: Your suggestion made so much sense in terms of not pinning boto3. I have updated the PR with the changes.

**Review by millerdev — APPROVED** [review]
I assume you're still working on the docs tests, but looks good so far.

**Review by gherceg — APPROVED** [review]

**Pull request merged**